VariantShredding.md

```diff
@@ -25,276 +25,302 @@
 
-# Parquet Layout
+| Value   | `metadata`       | `value`               | `typed_value` |
+|---------|------------------|-----------------------|---------------|
+| 34      | `01 00` v1/empty | null                  | `34`          |
+| null    | `01 00` v1/empty | `00` (null)           | null          |
+| "n/a"   | `01 00` v1/empty | `13 6E 2F 61` (`n/a`) | null          |
+| 100     | `01 00` v1/empty | null                  | `100`         |
 
-The `array` and `object` fields represent Variant array and object types, respectively.
-Arrays must use the three-level list structure described in https://github.com/apache/parquet-format/blob/master/LogicalTypes.md.
+Both `value` and `typed_value` are optional fields used together to encode a single value.
+Values in the two fields must be interpreted according to the following table:
 
-An `object` field must be a group.
-Each field name of this inner group corresponds to the Variant value's object field name.
-Each inner field's type is a recursively shredded variant value: that is, the fields of each object field must be one or more of `object`, `array`, `typed_value` or `variant_value`.
+| `value`  | `typed_value` | Meaning                                                  |
+|----------|---------------|----------------------------------------------------------|
+| null     | null          | The value is missing                                     |
+| non-null | null          | The value is present and may be any type, including null |
+| null     | non-null      | The value is present and is the shredded type            |
+| non-null | non-null      | The value is present and is a partially shredded object  |
 
-Similarly the elements of an `array` must be a group containing one or more of `object`, `array`, `typed_value` or `variant_value`.
+An object is _partially shredded_ when the `value` is an object and the `typed_value` is a shredded object.
 
-Each leaf in the schema can store an arbitrary Variant value.
-It contains an `variant_value` binary field and a `typed_value` field.
-If non-null, `variant_value` represents the value stored as a Variant binary.
-The `typed_value` field may be any type that has a corresponding Variant type.
-For each value in the data, at most one of the `typed_value` and `variant_value` may be non-null.
-A writer may omit either field, which is equivalent to all rows being null.
+If both fields are non-null and either is not an object, the value is invalid. Readers must either fail or return the `typed_value`.
 
-Dictionary IDs in a `variant_value` field refer to entries in the top-level `metadata` field.
+If a Variant is missing in a context where a value is required, readers must either fail or return a Variant null: basic type 0 (primitive) and physical type 0 (null).
+For example, if a Variant is required (like `measurement` above) and both `value` and `typed_value` are null, the returned `value` must be `00` (Variant null).
```

Comment:
As mentioned in my previous comment, I think it would be invalid for `measurement` to have both `value` and `typed_value` be null, and should be an error. I don't understand why we're recommend returning variant null as an option.

Reply:
This rule is to address the fact that arrays cannot contain a missing value. This is saying that if a value is required but both are null, the implementation must fill in a variant null.

Reply:
This rule(both value being null should be interpreted as json-null) is valid only for top level variant and array elements? I wonder how a top level variant can be inserted as both value and typed_value  being null if the top level field is required. That seems inconsistent. For arrays, it looks like we could also require value being variant encoded null(json null) rather than allowing both fields to be null.

Reply:
@sfc-gh-saya, if the top-level field is required but both fields are null, then the reader must produce a variant null value, `00`. We must state what happens in cases like this because it is possible for writers to produce them.